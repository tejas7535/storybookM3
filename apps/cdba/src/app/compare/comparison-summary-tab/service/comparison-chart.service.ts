import { Injectable } from '@angular/core';

import { TranslocoService } from '@jsverse/transloco';

@Injectable({ providedIn: 'root' })
export class ComparisonChartService {
  constructor(private readonly transloco: TranslocoService) {}

  readonly PURCHASE = this.transloco.translate(
    'compare.summary.common.purchase'
  );
  readonly REMAINDER = this.transloco.translate(
    'compare.summary.common.remainder'
  );
  readonly MOH = this.transloco.translate('compare.summary.common.moh');
  readonly BURDEN = this.transloco.translate('compare.summary.common.burden');
  readonly LABOUR = this.transloco.translate('compare.summary.common.labour');
  readonly RAW_MATERIAL = this.transloco.translate(
    'compare.summary.common.rawMaterial'
  );
  readonly TOTAL = this.transloco.translate('compare.summary.common.total');
}
