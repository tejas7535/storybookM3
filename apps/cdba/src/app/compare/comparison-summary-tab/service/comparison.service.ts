import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

import { Observable } from 'rxjs';

import { API, ComparisonPath } from '@cdba/shared/constants/api';
import { BomIdentifier } from '@cdba/shared/models';

import { Comparison } from '../../../shared/models/comparison.model';

@Injectable({ providedIn: 'root' })
export class ComparisonService {
  constructor(private readonly httpClient: HttpClient) {}

  getComparison(identifiers: BomIdentifier[]): Observable<Comparison> {
    const path = `${API.v1}/${ComparisonPath}`;

    return this.httpClient.post<Comparison>(path, identifiers);
  }
}
