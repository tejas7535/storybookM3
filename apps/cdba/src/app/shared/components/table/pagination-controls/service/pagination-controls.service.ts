/* eslint-disable @typescript-eslint/member-ordering */
// cannot have member-ordering with adjacent-overload-signatures rule simultaneously
import { Inject, Injectable } from '@angular/core';

import { LOCAL_STORAGE } from '@ng-web-apis/common';

import {
  MIN_PAGE_SIZE,
  PAGE_SIZE_OPTIONS,
  USER_PAGE_SIZE_KEY,
} from '@cdba/shared/constants/pagination';

@Injectable({ providedIn: 'root' })
export class PaginationControlsService {
  _currentPage = 0;
  _pages: number;
  _range: number;

  constructor(@Inject(LOCAL_STORAGE) private readonly localStorage: Storage) {}

  get currentPage(): number {
    return this._currentPage;
  }
  set currentPage(pageNumber: number) {
    this._currentPage = pageNumber;
  }
  get pages(): number {
    return this._pages;
  }
  set pages(pages: number) {
    this._pages = pages;
  }
  get range(): number {
    return this._range;
  }
  set range(range: number) {
    this._range = range;
  }

  getPageSizeFromLocalStorage(): number {
    const localStoragePageSize = Number(
      this.localStorage.getItem(USER_PAGE_SIZE_KEY)
    );

    if (
      localStoragePageSize !== undefined &&
      PAGE_SIZE_OPTIONS.includes(localStoragePageSize)
    ) {
      return localStoragePageSize;
    } else {
      this.localStorage.setItem(USER_PAGE_SIZE_KEY, String(MIN_PAGE_SIZE));

      return MIN_PAGE_SIZE;
    }
  }

  setPageSizeToLocalStorage(newPageSize: number): void {
    // Ignore values not from available options
    if (PAGE_SIZE_OPTIONS.includes(newPageSize)) {
      this.localStorage.setItem(USER_PAGE_SIZE_KEY, String(newPageSize));
    }
  }
}
