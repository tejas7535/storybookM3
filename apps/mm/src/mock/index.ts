import { BearinxDialogDescription } from '@caeonline/dynamic-forms';

import { environment } from '../environments/environment';
import * as dialog from './dialog.json';

const envDialogString = JSON.stringify(dialog).replace(
  /OPTIONS_URL/g,
  environment.baseUrl
);

const envDialog = JSON.parse(envDialogString);

export const DIALOG = (envDialog as any).default as BearinxDialogDescription;
