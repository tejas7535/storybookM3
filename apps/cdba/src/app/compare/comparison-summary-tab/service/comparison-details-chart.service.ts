import { Injectable } from '@angular/core';

import { TranslocoLocaleService } from '@jsverse/transloco-locale';
import { EChartsOption } from 'echarts';

import { DataPoint } from '@cdba/shared/components/bom-chart/data-point.model';
import { ComparisonType } from '@cdba/shared/constants';
import {
  CHART_HIGHLIGHT_ERROR,
  CHART_PRIMARY,
  CHART_SECONDARY,
  COST_SHARE_CATEGORY_COLORS,
} from '@cdba/shared/constants/colors';
import { ComparisonDetail } from '@cdba/shared/models/comparison.model';
import { CostShareService } from '@cdba/shared/services';

import { ComparisonChartService } from './comparison-chart.service';

@Injectable({ providedIn: 'root' })
export class ComparisonDetailsChartService {
  constructor(
    private readonly comparisonChartService: ComparisonChartService,
    private readonly localeService: TranslocoLocaleService,
    private readonly costShareService: CostShareService
  ) {}

  provideParetoChartConfig(
    details: ComparisonDetail[],
    currency: string
  ): EChartsOption {
    const xAxisData: string[] = [];
    const yAxisLineData: number[] = [];
    const yAxisBarDataPoints: DataPoint[] = [];
    let min = Number.POSITIVE_INFINITY;
    let max = Number.NEGATIVE_INFINITY;
    let cumulativePercentageCost = 0;

    details.forEach((detail) => {
      if (detail.costSharePercentage !== 0) {
        xAxisData.push(detail.title);
        min = Math.min(min, detail.totalCosts);
        max = Math.max(max, detail.totalCosts);

        cumulativePercentageCost += detail.costSharePercentage;
        cumulativePercentageCost = Number.parseFloat(
          cumulativePercentageCost.toPrecision(4)
        );
        yAxisLineData.push(cumulativePercentageCost);
        yAxisBarDataPoints.push({
          name: detail.title,
          value: detail.totalCosts,
          itemStyle: {
            color: COST_SHARE_CATEGORY_COLORS.get(
              this.costShareService.getCostShareCategory(
                detail.costSharePercentage / 100
              )
            ),
          },
        });
      }
    });

    return {
      tooltip: {
        trigger: 'axis',
        axisPointer: {
          type: 'cross',
          crossStyle: {
            color: '#999',
          },
        },
      },
      toolbox: {
        feature: {
          saveAsImage: { show: true },
        },
      },
      xAxis: [
        {
          type: 'category',
          data: xAxisData,
          axisPointer: {
            type: 'shadow',
          },
        },
      ],
      yAxis: [
        {
          type: 'value',
          min,
          max,
          axisLabel: {
            formatter: (value: string) => `${value} ${currency}`,
          },
        },
        {
          type: 'value',
          min: 0,
          max: 100,
          interval: 10,
          axisLabel: {
            formatter: (value: string) => `${value} %`,
          },
        },
      ],
      series: [
        {
          type: 'bar',
          tooltip: {
            valueFormatter: (value: number) =>
              `${this.localeService.localizeNumber(value, 'decimal')} ${currency}`,
          },
          data: yAxisBarDataPoints,
        },
        {
          type: 'line',
          color: CHART_HIGHLIGHT_ERROR,
          yAxisIndex: 1,
          tooltip: {
            valueFormatter: (value: string) => `${value} %`,
          },
          data: yAxisLineData,
        },
      ],
    } as EChartsOption;
  }

  provideDetailedBarChartConfig(
    detail: ComparisonDetail,
    firstMaterialDesignation: string,
    secondMaterialDesignation: string
  ): EChartsOption {
    const firstMaterialDesignationCosts: number[] = [];
    const secondMaterialDesignationCosts: number[] = [];

    detail.costDifferences.forEach((costDifference) => {
      switch (costDifference.type) {
        case ComparisonType.PURCHASE: {
          firstMaterialDesignationCosts[0] = costDifference.valueBom1;
          secondMaterialDesignationCosts[0] = costDifference.valueBom2;
          break;
        }
        case ComparisonType.RAW_MATERIAL: {
          firstMaterialDesignationCosts[1] = costDifference.valueBom1;
          secondMaterialDesignationCosts[1] = costDifference.valueBom2;
          break;
        }
        case ComparisonType.LABOUR: {
          firstMaterialDesignationCosts[2] = costDifference.valueBom1;
          secondMaterialDesignationCosts[2] = costDifference.valueBom2;
          break;
        }
        case ComparisonType.BURDEN: {
          firstMaterialDesignationCosts[3] = costDifference.valueBom1;
          secondMaterialDesignationCosts[3] = costDifference.valueBom2;
          break;
        }
        case ComparisonType.MOH: {
          firstMaterialDesignationCosts[4] = costDifference.valueBom1;
          secondMaterialDesignationCosts[4] = costDifference.valueBom2;
          break;
        }
        case ComparisonType.REMAINDER: {
          firstMaterialDesignationCosts[5] = costDifference.valueBom1;
          secondMaterialDesignationCosts[5] = costDifference.valueBom2;
          break;
        }
        case ComparisonType.TOTAL: {
          firstMaterialDesignationCosts[6] = costDifference.valueBom1;
          secondMaterialDesignationCosts[6] = costDifference.valueBom2;
          break;
        }
        default: {
          throw new Error(`Invalid Comparison Type: ${costDifference.type}`);
        }
      }
    });

    return {
      tooltip: {
        trigger: 'axis',
        axisPointer: {
          type: 'shadow',
        },
      },
      xAxis: {
        type: 'category',
        data: [
          this.comparisonChartService.PURCHASE,
          this.comparisonChartService.RAW_MATERIAL,
          this.comparisonChartService.LABOUR,
          this.comparisonChartService.BURDEN,
          this.comparisonChartService.MOH,
          this.comparisonChartService.REMAINDER,
          this.comparisonChartService.TOTAL,
        ],
        axisLabel: {
          interval: 0,
          width: 200,
        },
      },
      yAxis: {
        type: 'value',
      },
      series: [
        {
          type: 'bar',
          name: firstMaterialDesignation,
          data: firstMaterialDesignationCosts,
          itemStyle: {
            color: CHART_PRIMARY,
          },
        },
        {
          type: 'bar',
          name: secondMaterialDesignation,
          data: secondMaterialDesignationCosts,
          itemStyle: {
            color: CHART_SECONDARY,
          },
        },
      ],
    } as EChartsOption;
  }
}
