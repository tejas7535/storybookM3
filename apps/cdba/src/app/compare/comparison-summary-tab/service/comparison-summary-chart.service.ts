import { Injectable } from '@angular/core';

import { TranslocoService } from '@jsverse/transloco';
import { EChartsOption } from 'echarts/types/dist/echarts';

import {
  CHART_PRIMARY,
  CHART_SECONDARY,
  RADAR_CHART_REFERENCE_LINE,
} from '@cdba/shared/constants/colors';
import { ComparisonType } from '@cdba/shared/constants/comparison-type';
import {
  Comparison,
  ComparisonSummary,
  CostDifference,
} from '@cdba/shared/models/comparison.model';

import { ComparisonChartService } from './comparison-chart.service';

@Injectable({ providedIn: 'root' })
export class ComparisonSummaryChartService {
  constructor(
    private readonly transloco: TranslocoService,
    private readonly comparisonChartService: ComparisonChartService
  ) {}

  provideSummaryTableColumns(): string[] {
    return [
      'costType',
      'valueBom1',
      'valueBom2',
      'currency',
      'costDeltaValue',
      'costDeltaPercentage',
    ];
  }

  provideSummarizedRadarChartConfig(
    firstMaterialDesignation: string,
    secondMaterialDesignation: string,
    comparison: Comparison
  ): EChartsOption {
    const referenceLineTitle = this.transloco.translate(
      'compare.summary.common.referenceLine'
    );

    const purchaseCost = this.findCostDifferenceByComparisonType(
      comparison.summary,
      ComparisonType.PURCHASE
    );
    const remainderCost = this.findCostDifferenceByComparisonType(
      comparison.summary,
      ComparisonType.REMAINDER
    );
    const mohCost = this.findCostDifferenceByComparisonType(
      comparison.summary,
      ComparisonType.MOH
    );
    const burdenCost = this.findCostDifferenceByComparisonType(
      comparison.summary,
      ComparisonType.BURDEN
    );
    const labourCost = this.findCostDifferenceByComparisonType(
      comparison.summary,
      ComparisonType.LABOUR
    );
    const rawMaterialCost = this.findCostDifferenceByComparisonType(
      comparison.summary,
      ComparisonType.RAW_MATERIAL
    );

    // Calculate min max values
    let min = Number.POSITIVE_INFINITY;
    let max = Number.NEGATIVE_INFINITY;
    comparison.summary.costDifferences.forEach((costDifference) => {
      if (costDifference.type !== 'TOTAL') {
        min = Math.min(min, costDifference.valueBom1, costDifference.valueBom2);
        max = Math.max(max, costDifference.valueBom1, costDifference.valueBom2);
      }
    });
    min = Math.floor(min);
    max = Math.ceil(max);

    return {
      legend: {
        bottom: '0%',
        left: '5%',
        data: [
          referenceLineTitle,
          firstMaterialDesignation,
          secondMaterialDesignation,
        ],
      },
      tooltip: {
        trigger: 'item',
        valueFormatter: (value: string) => `${value} ${comparison.currency}`,
      },
      toolbox: {
        feature: {
          saveAsImage: { show: true },
        },
      },
      radar: {
        indicator: [
          {
            name: this.comparisonChartService.PURCHASE,
            min,
            max,
          },
          {
            name: this.comparisonChartService.REMAINDER,
            min,
            max,
          },
          {
            name: this.comparisonChartService.MOH,
            min,
            max,
          },
          {
            name: this.comparisonChartService.BURDEN,
            min,
            max,
          },
          {
            name: this.comparisonChartService.LABOUR,
            min,
            max,
          },
          {
            name: this.comparisonChartService.RAW_MATERIAL,
            min,
            max,
          },
        ],
      },
      series: [
        {
          type: 'radar',
          data: [
            {
              name: referenceLineTitle,
              value: [0, 0, 0, 0, 0, 0],
              itemStyle: {
                color: RADAR_CHART_REFERENCE_LINE,
              },
            },
            {
              name: firstMaterialDesignation,
              value: [
                purchaseCost.valueBom1,
                remainderCost.valueBom1,
                mohCost.valueBom1,
                burdenCost.valueBom1,
                labourCost.valueBom1,
                rawMaterialCost.valueBom1,
              ],
              itemStyle: {
                color: CHART_PRIMARY,
              },
            },
            {
              name: secondMaterialDesignation,
              value: [
                purchaseCost.valueBom2,
                remainderCost.valueBom2,
                mohCost.valueBom2,
                burdenCost.valueBom2,
                labourCost.valueBom2,
                rawMaterialCost.valueBom2,
              ],
              itemStyle: {
                color: CHART_SECONDARY,
              },
            },
          ],
        },
      ],
    } as EChartsOption;
  }

  findCostDifferenceByComparisonType(
    summary: ComparisonSummary,
    type: ComparisonType
  ): CostDifference {
    return summary.costDifferences.find(
      (costDifference) => costDifference.type === type
    );
  }
}
