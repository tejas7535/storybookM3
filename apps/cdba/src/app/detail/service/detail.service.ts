import { HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';

import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';

import { DataService } from '../../core/http/data.service';
import {
  BomIdentifier,
  BomItem,
  BomResult,
  ReferenceTypeIdModel,
  ReferenceTypeResultModel,
} from '../../core/store/reducers/detail/models';
import { CalculationsResultModel } from '../../core/store/reducers/detail/models/calculations-result-model';
import { Calculation } from '../../core/store/reducers/shared/models/calculation.model';

@Injectable({
  providedIn: 'root',
})
export class DetailService {
  private readonly DETAIL = 'detail';

  private readonly CALCULATIONS = 'calculations';

  private readonly PARAM_MATERIAL_NUMBER = 'material-number';
  private readonly PARAM_PLANT = 'plant';

  private readonly BOM = 'bom';
  private readonly PARAM_BOM_COSTING_DATE = 'bom-costing-date';
  private readonly PARAM_BOM_COSTING_NUMBER = 'bom-costing-number';
  private readonly PARAM_BOM_COSTING_TYPE = 'bom-costing-type';
  private readonly PARAM_BOM_COSTING_VERSION = 'bom-costing-version';
  private readonly PARAM_BOM_ENTERED_MANUALLY = 'bom-entered-manually';
  private readonly PARAM_BOM_REFERENCE_OBJECT = 'bom-reference-object';
  private readonly PARAM_BOM_VALUATION_VARIANT = 'bom-valuation-variant';

  private static defineBomTreeForAgGrid(
    items: BomItem[],
    idx: number
  ): BomItem[] {
    if (idx === items.length) {
      return items;
    }

    const currentItem = items[idx];
    const nextIdx = idx + 1;

    // special case level 1
    if (currentItem.level === 1) {
      currentItem.predecessorsInTree = [currentItem.materialDesignation];

      return DetailService.defineBomTreeForAgGrid(items, nextIdx);
    }

    const previousItem = items[idx - 1];

    if (previousItem.level < currentItem.level) {
      // previous item is parent
      currentItem.predecessorsInTree = [
        ...previousItem.predecessorsInTree,
        currentItem.materialDesignation,
      ];
    } else {
      // previous item has same parent or different parent
      const itemsToRemove = previousItem.level - currentItem.level + 1;
      const predecessors = [...previousItem.predecessorsInTree];

      // remove last items to have the actual parent as last item
      predecessors.splice(predecessors.length - itemsToRemove, itemsToRemove);

      // since duplicates are possible, add spaces to the key until it is unique
      let foundUniqueKey = false;
      let currentMaterialDesignation = currentItem.materialDesignation;
      while (!foundUniqueKey) {
        const temp = [...predecessors, currentMaterialDesignation];

        if (
          items
            .map((item) => item.predecessorsInTree)
            .find(
              (materialDesignations: string[]) =>
                materialDesignations.length === temp.length &&
                materialDesignations.every(
                  (value: string, index: number) => value === temp[index]
                )
            ) !== undefined
        ) {
          currentMaterialDesignation += ' ';
        } else {
          foundUniqueKey = true;
        }
      }

      currentItem.predecessorsInTree = [
        ...predecessors,
        currentMaterialDesignation,
      ];
    }

    return DetailService.defineBomTreeForAgGrid(items, nextIdx);
  }

  public constructor(private readonly dataService: DataService) {}

  public detail(
    item: ReferenceTypeIdModel
  ): Observable<ReferenceTypeResultModel> {
    const params: HttpParams = new HttpParams()
      .set(this.PARAM_MATERIAL_NUMBER, item.materialNumber)
      .set(this.PARAM_PLANT, item.plant);

    return this.dataService.getAll<ReferenceTypeResultModel>(
      this.DETAIL,
      params
    );
  }

  public calculations(materialNumber: string): Observable<Calculation[]> {
    const params = new HttpParams().set(
      this.PARAM_MATERIAL_NUMBER,
      materialNumber
    );

    return this.dataService
      .getAll<CalculationsResultModel>(this.CALCULATIONS, params)
      .pipe(map((response: CalculationsResultModel) => response.items));
  }

  public getBom(bomIdentifier: BomIdentifier): Observable<BomItem[]> {
    const params = new HttpParams()
      .set(this.PARAM_BOM_COSTING_DATE, bomIdentifier.bomCostingDate)
      .set(this.PARAM_BOM_COSTING_NUMBER, bomIdentifier.bomCostingNumber)
      .set(this.PARAM_BOM_COSTING_TYPE, bomIdentifier.bomCostingType)
      .set(this.PARAM_BOM_COSTING_VERSION, bomIdentifier.bomCostingVersion)
      .set(this.PARAM_BOM_ENTERED_MANUALLY, bomIdentifier.bomEnteredManually)
      .set(this.PARAM_BOM_REFERENCE_OBJECT, bomIdentifier.bomReferenceObject)
      .set(this.PARAM_BOM_VALUATION_VARIANT, bomIdentifier.bomValuationVariant);

    return this.dataService.getAll<BomResult>(this.BOM, params).pipe(
      map((response: BomResult) =>
        response.items.map((item) => ({
          ...item,
          predecessorsInTree: [],
        }))
      ),
      map((items: BomItem[]) => DetailService.defineBomTreeForAgGrid(items, 0))
    );
  }
}
