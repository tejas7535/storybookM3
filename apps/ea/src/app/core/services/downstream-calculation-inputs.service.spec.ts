import { DOWNSTREAM_API_RESULT_MOCK } from '@ea/testing/downstream-calculation-result.mock';
import { TranslocoService } from '@jsverse/transloco';
import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import { DownstreamAPIResponse } from './downstream-calculation.service.interface';
import { DownstreamCalculationInputsService } from './downstream-calculation-inputs.service';

describe('DownstreamCalculationInputsService', () => {
  let spectator: SpectatorService<DownstreamCalculationInputsService>;
  let translocoService: TranslocoService;
  const createService = createServiceFactory({
    service: DownstreamCalculationInputsService,

    imports: [provideTranslocoTestingModule({ en: {} })],
  });

  beforeEach(() => {
    spectator = createService();
    translocoService = spectator.inject(TranslocoService);
    translocoService.translate = jest
      .fn()
      .mockImplementation((key: string) => key);
  });

  it('should create the service', () => {
    expect(spectator.service).toBeTruthy();
  });

  it('should format downstream inputs correctly', () => {
    const downstreamResponse: DownstreamAPIResponse = {
      ...DOWNSTREAM_API_RESULT_MOCK,
    };

    const result = spectator.service.formatDownstreamInputs(downstreamResponse);
    expect(result).toMatchSnapshot();
  });

  it('should format downstream fossil energy source inputs', () => {
    const downstreamResponse: DownstreamAPIResponse = {
      ...DOWNSTREAM_API_RESULT_MOCK,
      inputData: {
        ...DOWNSTREAM_API_RESULT_MOCK.inputData,
        operatingConditions: {
          ...DOWNSTREAM_API_RESULT_MOCK.inputData.operatingConditions,
          emissionFactor: 'LB_FOSSIL_ENERGY',
        },
      },
    };

    const result = spectator.service.formatDownstreamInputs(downstreamResponse);
    expect(result).toMatchSnapshot();
  });

  it('should create translation keys correctly', () => {
    const basePath = 'calculationResultReport.downstreamInputs.otherConditions';
    const keys = [
      'title',
      'temperature',
      'lubricationMethod',
      'greaseType',
      'operatingTimeInHours',
    ];
    const expectedKeys = {
      title: 'calculationResultReport.downstreamInputs.otherConditions.title',
      temperature:
        'calculationResultReport.downstreamInputs.otherConditions.temperature',
      lubricationMethod:
        'calculationResultReport.downstreamInputs.otherConditions.lubricationMethod',
      greaseType:
        'calculationResultReport.downstreamInputs.otherConditions.greaseType',
      operatingTimeInHours:
        'calculationResultReport.downstreamInputs.otherConditions.operatingTimeInHours',
    };

    const result = (spectator.service as any).createTranslationKeys(
      basePath,
      keys
    );
    expect(result).toEqual(expectedKeys);
  });
});
