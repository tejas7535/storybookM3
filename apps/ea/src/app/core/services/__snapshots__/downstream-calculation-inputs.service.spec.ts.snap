// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownstreamCalculationInputsService should format downstream fossil energy source inputs 1`] = `
Object {
  "bearing": Object {
    "hasNestedStructure": false,
    "subItems": Array [
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.bearing.designation",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "NKI38/20-XL",
      },
      Object {
        "abbreviation": "d",
        "designation": "calculationResultReport.downstreamInputs.bearing.innerDiameter",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": 38,
      },
      Object {
        "abbreviation": "D",
        "designation": "calculationResultReport.downstreamInputs.bearing.outerDiameter",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": 53,
      },
      Object {
        "abbreviation": "B",
        "designation": "calculationResultReport.downstreamInputs.bearing.width",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": 20,
      },
      Object {
        "abbreviation": "C",
        "designation": "calculationResultReport.downstreamInputs.bearing.dynamicLoadRating",
        "hasNestedStructure": false,
        "unit": "N",
        "value": 30500,
      },
      Object {
        "abbreviation": "C0",
        "designation": "calculationResultReport.downstreamInputs.bearing.staticLoadRating",
        "hasNestedStructure": false,
        "unit": "N",
        "value": 51000,
      },
      Object {
        "abbreviation": "Cu",
        "designation": "calculationResultReport.downstreamInputs.bearing.fatigueLoadLimit",
        "hasNestedStructure": false,
        "unit": "N",
        "value": 9100,
      },
      Object {
        "abbreviation": "n_lim_o",
        "designation": "calculationResultReport.downstreamInputs.bearing.limitingSpeedOil",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": 10700,
      },
      Object {
        "abbreviation": "n_lim_g",
        "designation": "calculationResultReport.downstreamInputs.bearing.limitingSpeedGrease",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": 6400,
      },
    ],
    "title": "calculationResultReport.downstreamInputs.bearing.bearingTitle",
  },
  "load": Object {
    "hasNestedStructure": true,
    "loadCases": Array [
      Object {
        "hasNestedStructure": false,
        "subItems": Array [
          Object {
            "abbreviation": "q",
            "designation": "calculationResultReport.downstreamInputs.load.timePortion",
            "hasNestedStructure": false,
            "unit": "%",
            "value": 100,
          },
          Object {
            "abbreviation": undefined,
            "designation": "operationConditions.rotatingCondition.typeOfMotion",
            "hasNestedStructure": false,
            "unit": undefined,
            "value": "operationConditions.rotatingCondition.options.LB_ROTATING",
          },
          Object {
            "abbreviation": "n_i",
            "designation": "calculationResultReport.downstreamInputs.load.speed",
            "hasNestedStructure": false,
            "unit": "1/min",
            "value": 950,
          },
          Object {
            "abbreviation": "T",
            "designation": "calculationResultReport.downstreamInputs.load.operatingTemperature",
            "hasNestedStructure": false,
            "unit": "°C",
            "value": 70,
          },
          Object {
            "abbreviation": "Fa",
            "designation": "calculationResultReport.downstreamInputs.load.axialLoad",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "0",
          },
          Object {
            "abbreviation": "Fr",
            "designation": "calculationResultReport.downstreamInputs.load.radialLoad",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "950",
          },
        ],
        "title": "",
      },
    ],
    "title": "calculationResultReport.downstreamInputs.load.loadTitle",
  },
  "operatingConditions": Object {
    "baseConditions": Array [
      Object {
        "abbreviation": "t",
        "designation": "calculationResultReport.downstreamInputs.otherConditions.temperature",
        "hasNestedStructure": false,
        "unit": "°C",
        "value": 20,
      },
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.otherConditions.lubricationMethod",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "grease",
      },
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.otherConditions.greaseType",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "Arcanol MULTI2",
      },
    ],
    "downstreamConditions": Array [
      Object {
        "abbreviation": undefined,
        "designation": "operationConditions.energySource.title",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "operationConditions.energySource.options.fossil",
      },
      Object {
        "abbreviation": undefined,
        "designation": "operationConditions.energySource.fossilOrigin",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "operationConditions.energySource.fossilOriginOptions.e10",
      },
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.otherConditions.operatingTimeInHours",
        "hasNestedStructure": false,
        "unit": "h",
        "value": 8766,
      },
    ],
    "hasNestedStructure": false,
    "title": "calculationResultReport.downstreamInputs.otherConditions.conditionsTitle",
  },
}
`;

exports[`DownstreamCalculationInputsService should format downstream inputs correctly 1`] = `
Object {
  "bearing": Object {
    "hasNestedStructure": false,
    "subItems": Array [
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.bearing.designation",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "NKI38/20-XL",
      },
      Object {
        "abbreviation": "d",
        "designation": "calculationResultReport.downstreamInputs.bearing.innerDiameter",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": 38,
      },
      Object {
        "abbreviation": "D",
        "designation": "calculationResultReport.downstreamInputs.bearing.outerDiameter",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": 53,
      },
      Object {
        "abbreviation": "B",
        "designation": "calculationResultReport.downstreamInputs.bearing.width",
        "hasNestedStructure": false,
        "unit": "mm",
        "value": 20,
      },
      Object {
        "abbreviation": "C",
        "designation": "calculationResultReport.downstreamInputs.bearing.dynamicLoadRating",
        "hasNestedStructure": false,
        "unit": "N",
        "value": 30500,
      },
      Object {
        "abbreviation": "C0",
        "designation": "calculationResultReport.downstreamInputs.bearing.staticLoadRating",
        "hasNestedStructure": false,
        "unit": "N",
        "value": 51000,
      },
      Object {
        "abbreviation": "Cu",
        "designation": "calculationResultReport.downstreamInputs.bearing.fatigueLoadLimit",
        "hasNestedStructure": false,
        "unit": "N",
        "value": 9100,
      },
      Object {
        "abbreviation": "n_lim_o",
        "designation": "calculationResultReport.downstreamInputs.bearing.limitingSpeedOil",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": 10700,
      },
      Object {
        "abbreviation": "n_lim_g",
        "designation": "calculationResultReport.downstreamInputs.bearing.limitingSpeedGrease",
        "hasNestedStructure": false,
        "unit": "1/min",
        "value": 6400,
      },
    ],
    "title": "calculationResultReport.downstreamInputs.bearing.bearingTitle",
  },
  "load": Object {
    "hasNestedStructure": true,
    "loadCases": Array [
      Object {
        "hasNestedStructure": false,
        "subItems": Array [
          Object {
            "abbreviation": "q",
            "designation": "calculationResultReport.downstreamInputs.load.timePortion",
            "hasNestedStructure": false,
            "unit": "%",
            "value": 100,
          },
          Object {
            "abbreviation": undefined,
            "designation": "operationConditions.rotatingCondition.typeOfMotion",
            "hasNestedStructure": false,
            "unit": undefined,
            "value": "operationConditions.rotatingCondition.options.LB_ROTATING",
          },
          Object {
            "abbreviation": "n_i",
            "designation": "calculationResultReport.downstreamInputs.load.speed",
            "hasNestedStructure": false,
            "unit": "1/min",
            "value": 950,
          },
          Object {
            "abbreviation": "T",
            "designation": "calculationResultReport.downstreamInputs.load.operatingTemperature",
            "hasNestedStructure": false,
            "unit": "°C",
            "value": 70,
          },
          Object {
            "abbreviation": "Fa",
            "designation": "calculationResultReport.downstreamInputs.load.axialLoad",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "0",
          },
          Object {
            "abbreviation": "Fr",
            "designation": "calculationResultReport.downstreamInputs.load.radialLoad",
            "hasNestedStructure": false,
            "unit": "N",
            "value": "950",
          },
        ],
        "title": "",
      },
    ],
    "title": "calculationResultReport.downstreamInputs.load.loadTitle",
  },
  "operatingConditions": Object {
    "baseConditions": Array [
      Object {
        "abbreviation": "t",
        "designation": "calculationResultReport.downstreamInputs.otherConditions.temperature",
        "hasNestedStructure": false,
        "unit": "°C",
        "value": 20,
      },
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.otherConditions.lubricationMethod",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "grease",
      },
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.otherConditions.greaseType",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "Arcanol MULTI2",
      },
    ],
    "downstreamConditions": Array [
      Object {
        "abbreviation": undefined,
        "designation": "operationConditions.energySource.title",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "operationConditions.energySource.options.electricity",
      },
      Object {
        "abbreviation": undefined,
        "designation": "operationConditions.energySource.electricityRegion",
        "hasNestedStructure": false,
        "unit": undefined,
        "value": "operationConditions.energySource.electricityRegionOption.europeanUnion",
      },
      Object {
        "abbreviation": undefined,
        "designation": "calculationResultReport.downstreamInputs.otherConditions.operatingTimeInHours",
        "hasNestedStructure": false,
        "unit": "h",
        "value": 8766,
      },
    ],
    "hasNestedStructure": false,
    "title": "calculationResultReport.downstreamInputs.otherConditions.conditionsTitle",
  },
}
`;
